src/components/organisms/main/ranking/RankingSection.tsx

```diff
@@ -9,14 +9,23 @@ import Button from '@components/atoms/button/Button';
 import {
   RankingSectionTitle, RankingSectionTitleContainer,
 } from '@components/organisms/main/ranking/RankingSection.styles';
+import FetchStatusBoundary
+  from '@components/atoms/container/FetchStatusBoundary';
 import { TargetFilter, RankFilter } from '@/types';
 
 function RankingSection() {
   const [targetFilter, setTargetFilter] = useState<TargetFilter>('ALL');
   const [rankFilter, setRankFilter] = useState<RankFilter>('MANY_WISH');
   const [isFolded, setIsFolded] = useState(true);
 
-  const products = useFetchProducts({ targetFilter, rankFilter });
+  const { products, fetchStatus } = useFetchProducts({ targetFilter, rankFilter });
+
+  const DISPLAY_COUNT_WHEN_FOLDED = 6;
+
+  const isButtonEnabled = useCallback(
+    () => products.length >= DISPLAY_COUNT_WHEN_FOLDED,
+    [products],
+  );
```

Comment:
로컬에서 실행해봤을때 `products`가 `undefined` 인 케이스 처리가 안되어 있어 에러가 발생합니다.
혹시 DISPLAY_COUNT_WHEN_FOLDED 값을 초과했을때 해당 버튼이 노출되어야 하는게 아닐까요?
또 아래 동작을 보면 버튼을 노출할지 여부로 보여요~

```suggestion
  const showButton = useCallback(
    () => products?.length > DISPLAY_COUNT_WHEN_FOLDED,
    [products],
  );
```

Reply:
아하, enabled는 버튼의 활성화/비활성화 속성과 관련이 있어 혼동될 수 있겠네요..!
조건을 제가 원하는대로 작동하도록 수정하고 함수명을 적절히 바꿔봤습니다!